radix-engine/src/engine/wasm_runtime.rs

```diff
@@ -15,41 +15,41 @@ use crate::engine::SystemApi;
 use crate::model::Component;
 use crate::wasm::*;
 
-pub struct RadixEngineWasmRuntime<'s, S, W, I>
+use super::CostUnitCounter;
+
+pub struct RadixEngineWasmRuntime<'s, 'c, S, W, I>
 where
     S: SystemApi<W, I>,
     W: WasmEngine<I>,
     I: WasmInstance,
 {
     this: ScryptoActorInfo,
     system_api: &'s mut S,
-    tbd_limit: u32,
-    tbd_balance: u32,
+    cost_unit_counter: &'c mut CostUnitCounter,
     phantom1: PhantomData<W>,
     phantom2: PhantomData<I>,
 }
 
-impl<'s, S, W, I> RadixEngineWasmRuntime<'s, S, W, I>
+impl<'s, 'c, S, W, I> RadixEngineWasmRuntime<'s, 'c, S, W, I>
 where
     S: SystemApi<W, I>,
     W: WasmEngine<I>,
     I: WasmInstance,
 {
-    pub fn new(this: ScryptoActorInfo, system_api: &'s mut S, tbd_limit: u32) -> Self {
+    pub fn new(
+        this: ScryptoActorInfo,
+        system_api: &'s mut S,
+        cost_unit_counter: &'c mut CostUnitCounter,
```

Comment:
Does it make more sense for the runtime to own the counter?

Reply:
It does because both WASM and RE need to update the counter. Next PR :)